cases/file_to_s3_reliability/README.md

```diff
@@ -0,0 +1,25 @@
+# File to S3 Reliability Test
+
+This is a long-running test of Vector tailing a file and sending the data to S3.
+It uses the [`verifiable-logger`][0] project both to generate the log data and
+verify that it all reaches S3.
+
+## Try it
+
+You can run this test via:
+
+```
+test -t file_to_s3_reliability
```

Comment:
One thing missing here: how do we shut down the test and destroy all the resources?

Reply:
Currently, we kill the VMs via CloudWatch alert. We need a better way, maybe could use lambda for it.

Reply:
Yes, we should add a `bin/teardown -t <test> -c <config>`. The `terraform destroy` command should make this easy. **It should only be called for the local test state, not global state, obviously.**

> Currently, we kill the VMs via CloudWatch alert. We need a better way, maybe could use lambda for it.

What's wrong with CloudWatch? I think it works quite well for this.

Reply:
CloudWatch works!
But if we create resources per test - then if we have `bin/teardown`, we have to invoke it with enough context so that it only deletes resources associated with a particular test.

So this is where we use lambda: we can assign the invocation of `bin/teardown` with all required context delayed by, let's say, three hours. We will then schedule this lambda invocation as part of `bin/test` run. It'll allow us to not only clean up the VMs, but also all the associated terraform state - policies, S3 buckets, VPC, and everything else that we create per-test.

I think this solution can replace our CloudWatch VMs removal because we can clean up everything, including the VMs!

Reply:
This should solve it for now: https://github.com/timberio/vector-test-harness/pull/52